### PR TITLE
FAI-416 - TrustyAI explainability benchmarks

### PR DESCRIPTION
**Pull request by tteofili**
Enable `pytest-benchmark` level benchmarks for TrustyAI metrics on common models for continuous explanation quality evaluation.

### CODE REVIEW

File: src/trustyai/metrics/saliency.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+from trustyai.model import simple_prediction
+from trustyai.explainers.lime import LimeExplainer
+from trustyai.explainers.shap import SHAPExplainer
+from trustyai.explainers import LocalExplainer
+
+from . import ExplainabilityMetrics
+
+
+def impact_score(model, pred_input, explainer, k, is_model_callable=False):
```
Comment (ruivieira) [inline src/trustyai/metrics/saliency.py:15]: Perhaps we could add argument typing in these signatures.

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+pyarrow
 numpydoc
-matplotlib==3.5.1
-pandas==1.2.5
+matplotlib
+pandas
 pytest-xdist==2.5.0
 pytest-benchmark
-bokeh==2.4.3
+bokeh
```
Comment (ruivieira) [inline requirements-dev.txt:19]: I think we should leave the versions pinned.

File: src/trustyai/metrics/saliency.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    chunk_size: int
+        the chunk of inputs to fetch fro the distribution
+
+    Returns
+    -------
+    :float:
+        the local saliency f1 metric
+    """
+    if isinstance(explainer, LimeExplainer):
+        local_explainer = LocalExplainer(explainer._explainer)
```
Comment (ruivieira) [inline src/trustyai/metrics/saliency.py:126]: I haven't tested it, but I wonder if casting to the interface directly with JPype would remove the need to re-implement the `LocalExplainer` "type".

For instance:

```python
from org.kie.trustyai.explainability.local import LocalExplainer

local_explainer = JObject(explainer._explainer, LocalExplainer)
```

**Review by ruivieira — APPROVED** [review]

File: src/trustyai/metrics/saliency.py (hunk 1 above)
Comment (tteofili) [reply src/trustyai/metrics/saliency.py:15]: sure

File: src/trustyai/metrics/saliency.py (hunk 3 above)
Comment (tteofili) [reply src/trustyai/metrics/saliency.py:126]: well, thanks  @ruivieira that's much better and avoids introducing extra class needlessly.

**Comment by tteofili** [comment]
> LGTM, thanks.
> 
> The only thing I think should change is not removing the version pinning.

yeah, that was an unwanted change, I'll revert that.

**Review by ruivieira — APPROVED** [review]

**Pull request merged**